File: src/chessgames/chess/Position.java
package chessgames.chess;

public class Position {
    private int xPos;
    private int yPos;
    private int score = 0;

    public Position(int x, int y) {
        this.xPos = x;
        this.yPos = y;
    }

    public Position(int x, int y, int score) {
        this.xPos = x;
        this.yPos = y;
        this.score = score;
    }

    public int getxPos() {
        return xPos;
    }

    public int getyPos() {
        return yPos;
    }

    public int getScore() {
        return score;
    }
}

File: src/Position.cpp
#include "Position.h"

Position::Position(int x, int y) {
    this->x = x;
    this->y = y;
}

Position Position::of(int x, int y) {
    return Position(x, y);
}

bool Position::equals(Position p) {
    return this->x == p.x && this->y == p.y;
}
